platform/mbed_sleep.h

```diff
@@ -64,6 +64,34 @@ extern "C" {
  * }
  * @endcode
  */
+#ifdef MBED_SLEEP_STATS_ENABLED
+
+#define sleep_manager_lock_deep_sleep()           \
+    do                                            \
+    {                                             \
+        sleep_manager_lock_deep_sleep_internal(); \
+        sleep_tracker_lock(__FILE__, __LINE__);   \
```

Comment:
You can use __MODULE__ for ARM and __BASE_FILE__ for GCC_ARM instead of __FILE__ to get just filename without path details.

--no_path_in_file_macros option in IAR is to exclude path from __BASE_FILE__, but I have not tested that.